src/environment.cpp

```diff
@@ -2116,8 +2117,7 @@ void ClientEnvironment::step(float dtime)
 	*/
 
 	u32 loopcount = 0;
-	do
-	{
+	while (dtime_downcount > 0.001) {
```

Comment:
Replacing `do {} while()` with `while () {}` changes behaviour. Have you taken that into account?